libmisc/user_busy.c

```diff
@@ -123,26 +171,29 @@ static int check_status (const char *name, const char *sname, uid_t uid)
 	while (fgets (line, sizeof (line), sfile) == line) {
 		if (strncmp (line, "Uid:\t", 5) == 0) {
 			unsigned long ruid, euid, suid;
+
 			assert (uid == (unsigned long) uid);
+			(void) fclose (sfile);
 			if (sscanf (line,
 			            "Uid:\t%lu\t%lu\t%lu\n",
 			            &ruid, &euid, &suid) == 3) {
 				if (   (ruid == (unsigned long) uid)
 				    || (euid == (unsigned long) uid)
-				    || (suid == (unsigned long) uid)
+				    || (suid == (unsigned long) uid) ) {
+					return 1;
+				}
 #ifdef ENABLE_SUBIDS
-				    || have_sub_uids(name, ruid, 1)
-				    || have_sub_uids(name, euid, 1)
-				    || have_sub_uids(name, suid, 1)
-#endif				/* ENABLE_SUBIDS */
+				if (  (   have_sub_uids(name, ruid, 1)
+				       || have_sub_uids(name, euid, 1)
+				       || have_sub_uids(name, suid, 1))
+                                    && in_mapping(sname, ruid, euid, suid)
```

Comment:
The indentation looks a bit off here. Maybe using tabs instead of spaces?

Reply:
Fixed, thank you.